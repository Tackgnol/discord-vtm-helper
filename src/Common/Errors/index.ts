export * from './FirebaseError';
export * from './InvalidInputError';
