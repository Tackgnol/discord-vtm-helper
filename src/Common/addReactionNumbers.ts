import { Message } from 'discord.js';
import { reactionNumbers } from './reactionNumbers';

export const addReactionNumbers = async (message: Message) => {
	await message.react(reactionNumbers[0]);
	await message.react(reactionNumbers[1]);
	await message.react(reactionNumbers[2]);
	await message.react(reactionNumbers[3]);
	await message.react(reactionNumbers[4]);
	await message.react(reactionNumbers[5]);
	await message.react(reactionNumbers[6]);
	await message.react(reactionNumbers[7]);
	await message.react(reactionNumbers[8]);
	await message.react(reactionNumbers[9]);
};
