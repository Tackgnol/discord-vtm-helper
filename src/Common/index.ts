export * from './adminMessages';
export * from './isAdminCommand';
export * from './parseCommandMessage';
export * from './parseEventMessage';
export * from './Play';
export * from './reactionNumbers';
export * from './RichEmbeds';
