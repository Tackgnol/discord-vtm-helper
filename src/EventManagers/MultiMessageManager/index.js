const { RichEmbed } = require('discord.js');
const settings = require('../../../config/settings');
const { isEmpty, isNil } = require('lodash');

class MultiMessageManager {
	constructor(message, client = null) {
		if (isNil(message.client)) {
			this.message = message;
			this.client = client;
		} else {
			this.message = message;
			this.client = message.client;
		}
	}

	messageUsers(messageObject) {
		if (isEmpty(messageObject)) {
			return;
		}
		const messageChanel = this.message.channel;
		if (messageChanel.type === 'text') {
			messageObject.forEach(m => {
				const channelMembers = messageChanel.members;
				const userList = m.userList;
				if (!isEmpty(userList)) {
					userList.forEach(u => {
						const foundUser = channelMembers.find(
							val => val.user.username === u
						);
						if (!isNil(foundUser)) {
							const richEmbed = new RichEmbed()
								.setColor(settings.colors.richEmbeddedMain)
								.setTitle(settings.Lines.userMessageHeader)
								.setDescription(m.value)
								.setColor(settings.colors.richEmbeddedMain);
							foundUser.send(richEmbed);
						}
					});
				}
			});
		}
	}
}

module.exports = MultiMessageManager;
