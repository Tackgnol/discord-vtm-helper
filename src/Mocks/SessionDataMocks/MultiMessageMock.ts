import { IMultiPlayerMessage } from '../../Models/GameData';

export const multiMessageMock: IMultiPlayerMessage = {
	userList: ['0'],
	name: 'test',
	message: "you've got mail",
};
