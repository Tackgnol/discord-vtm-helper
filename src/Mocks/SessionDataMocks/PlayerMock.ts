import { IPlayer } from '../../Models/GameData';
import { testNPCWithFacts, testNPCWithoutFacts } from './NPCMocks';

export const playerMock: IPlayer = {
	npcSet: [testNPCWithFacts, testNPCWithoutFacts],
	discordUserName: 'test',
	statisticsSet: [{ name: 'test', value: 2 }],
	id: '0',
};
