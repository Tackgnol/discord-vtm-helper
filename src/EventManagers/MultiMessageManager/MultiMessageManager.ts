import { MessageEmbed } from 'discord.js';
import { settings } from '../../config/settings';
import { isEmpty, isNil } from 'lodash';
import { MessageList, Reply, ReplyType } from '../../Models/AppModels';
import { InvalidInputError } from '../../Common/Errors/InvalidInputError';

class MultiMessageManager {
	messageUsers(users: string[], channelMembers: string[], value: string): Reply {
		if (isEmpty(users)) {
			throw new InvalidInputError('This channel has no users!');
		}
		const messages: MessageList[] = [];
		if (!isEmpty(users)) {
			users.forEach(u => {
				const foundUser = channelMembers.find(val => val === u);
				if (!isNil(foundUser)) {
					const richEmbed = new MessageEmbed()
						.setColor(settings.colors.richEmbeddedMain)
						.setTitle(settings.lines.userMessageHeader)
						.setDescription(value)
						.setColor(settings.colors.richEmbeddedMain);
					messages.push({ message: richEmbed, recipient: u });
				}
			});
			return { type: ReplyType.Multi, value: messages };
		}
		throw new InvalidInputError('Channel has no users');
	}
}

export default MultiMessageManager;
