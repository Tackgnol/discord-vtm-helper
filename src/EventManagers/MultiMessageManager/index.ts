import { Collection, GuildMember, MessageEmbed, User } from 'discord.js';
import { settings } from '../../config/settings';
import { isEmpty, isNil } from 'lodash';
import { IMessageList, IReply, ReplyType } from '../../Models/AppModels';
import { InvalidInputError } from '../../Common/Errors/InvalidInputError';

class MultiMessageManager {
	messageUsers(users: User[], channelMembers: Collection<string, GuildMember>, value: string): IReply {
		if (isEmpty(users)) {
			throw new InvalidInputError('This channel has no users!');
		}
		const messages: IMessageList[] = [];
		if (!isEmpty(users)) {
			users.forEach(u => {
				const foundUser = channelMembers.find(val => val.user.username === u.username);
				if (!isNil(foundUser)) {
					const richEmbed = new MessageEmbed()
						.setColor(settings.colors.richEmbeddedMain)
						.setTitle(settings.lines.userMessageHeader)
						.setDescription(value)
						.setColor(settings.colors.richEmbeddedMain);
					messages.push({ message: richEmbed, recipient: u });
				}
			});
		}
		return { type: ReplyType.Multi, value: messages };
	}
}

export default MultiMessageManager;
