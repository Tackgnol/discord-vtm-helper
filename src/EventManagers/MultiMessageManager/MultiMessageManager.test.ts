import MultiMessageManager from './MultiMessageManager';
import { multiMessageMock } from '../../Mocks/SessionDataMocks/MultiMessageMock';
import { playerMock } from '../../Mocks/SessionDataMocks/PlayerMock';
import { expect } from 'chai';
import { InvalidInputError } from '../../Common/Errors';

describe('EventManagers >> MultiMessageManager >> MultiMessageManager', () => {
	const manager = new MultiMessageManager();
	it('Returns a message set for users', () => {
		const { message, userList } = multiMessageMock;
		const result = manager.messageUsers([playerMock.id], userList, message);
		expect(result).toMatchSnapshot();
		expect(result.value).to.have.lengthOf(1);
	});

	it('Throws an error if no users are in the channel', () => {
		const { message } = multiMessageMock;
		expect(() => {
			manager.messageUsers([playerMock.id], [], message);
		}).to.throw(InvalidInputError);
	});

	it('Throws an error if no users to send to', () => {
		const { message, userList } = multiMessageMock;
		expect(() => {
			manager.messageUsers([], userList, message);
		}).to.throw(InvalidInputError);
	});
});
