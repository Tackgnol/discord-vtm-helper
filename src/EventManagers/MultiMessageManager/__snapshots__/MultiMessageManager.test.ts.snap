// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventManagers >> MultiMessageManager >> MultiMessageManager Returns a message set for users 1`] = `
Object {
  "type": 2,
  "value": Array [
    Object {
      "message": Object {
        "author": null,
        "color": 9044739,
        "description": "you've got mail",
        "fields": Array [],
        "footer": null,
        "image": null,
        "thumbnail": null,
        "timestamp": null,
        "title": "Whispers in the dark... be it friend or foe or thyself?",
        "type": "rich",
        "url": null,
      },
      "recipient": "0",
    },
  ],
}
`;
